ui/src/views/DimensionEdit.vue

```diff
@@ -45,7 +48,33 @@ export default class extends Vue {
 
   async mounted (): Promise<void> {
     if (this.operation) {
-      this.shape = await api.fetchOperationShape(this.operation)
+      const shape = await api.fetchOperationShape(this.operation)
+
+      if (!shape) throw new Error('Shape not found')
+
+      // Manually build the dimensions list because `addList` causes a blank
+      // node clash
+      const dimensionsList = shape.pointer.blankNode('_dimensions-0')
+      let position: GraphPointer<any> = dimensionsList
+      this.dimensions.forEach((d, index) => {
+        position
+          .addOut(rdf.first, d.about, dimPtr => {
+            dimPtr.addOut(schema.name, d.name)
+          })
+          .addOut(rdf.rest, position.blankNode(`_dimensions-${index + 1}`))
+
+        position = position.out(rdf.rest)
+      })
+      position.in(rdf.rest).deleteOut(rdf.rest).addOut(rdf.rest, rdf.nil)
```

Comment:
I had to do this horror to avoid a weird blank node clash when using `addList`. We may want to fix that, but I wanted to get it working first.